### PR TITLE
Add recaptcha to invitation modal [SCI-2935]

### PR DESCRIPTION
**Pull request by urbanrotnik**
Jira ticket: [SCI-2935](https://biosistemika.atlassian.net/browse/SCI-2935?filter=-1)

**todo:**
Add ReCaptcha to invitation modal 

**What's done:** 
Added `before_filter`, updated `ajax` call `invite_users`

**NOTE**
I didn't use existing implementation with `recaptcha_tag` helper because I am not inside of `form` element. 
Now we have included `recaptcha` at 3 diffrent users experiences, everyone has own `before_filter` for checking captcha. If we will use `recaptcha` on more places, implementing universal `check_captcha` on higher controller level should be a better way and keep other logic after validation outside of that `before_filter`.

### CODE REVIEW

File: app/controllers/users/invitations_controller.rb [hunk 1]
```diff
@@ -188,6 +190,14 @@ def generate_notification(user, target_user, role, team)
       end
     end
 
+    def check_captcha_for_invite
+      if Rails.configuration.x.enable_recaptcha
+        unless verify_recaptcha
+          render json: { recaptcha_error: t('invite_users.errors.recaptcha') }
```
Comment (radiokills) [inline app/controllers/users/invitations_controller.rb]: would it make sense to send a status code with this (i.e. 400)?

File: app/assets/javascripts/users/settings/teams/invite_users_modal.js [hunk 2]
```diff
@@ -123,23 +137,31 @@
...
-            stepResultsDiv.html(data.html);
-            stepResults.show();
-            // Add 'data-invited="true"' status to modal element
-            modal.attr('data-invited', 'true');
+            if (data.recaptcha_error) {
+              animateSpinner(modalDialog, false);
+              recaptchaErrorMsgDiv.removeClass('hidden');
+              recaptchaErrorText.text(data.recaptcha_error);
+              reloadRecaptchaIfExists();
+            } else {
```
Comment (radiokills) [inline app/assets/javascripts/users/settings/teams/invite_users_modal.js]: In the ideal world this should be handled by error callback. For this controller should return an HTTP error code (for example 422)

File: app/controllers/users/invitations_controller.rb (hunk 1 above)
Comment (urbanrotnik) [reply app/controllers/users/invitations_controller.rb]: 👍 Yeap

File: app/assets/javascripts/users/settings/teams/invite_users_modal.js (hunk 2 above)
Comment (urbanrotnik) [reply app/assets/javascripts/users/settings/teams/invite_users_modal.js]: Agree 👍

**Pull request merged**